alf/algorithms/sac_algorithm.py

```diff
@@ -784,10 +784,16 @@ def _critic_train_step(self, inputs: TimeStep, state: SacCriticState,
         return state, info
 
     def _alpha_train_step(self, log_pi):
-        alpha_loss = nest.map_structure(
-            lambda la, lp, t: la * (-lp - t).detach(), self._log_alpha, log_pi,
-            self._target_entropy)
-        return sum(nest.flatten(alpha_loss))
+        if self._act_type == ActionType.Mixed:
+            alpha_loss = nest.map_structure(
+                lambda la, lp, t: la * (-lp - t).detach(), self._log_alpha,
```

Comment:
this still have problem if the continuous action is nested

Reply:
Now also fixed.